src/com/xilinx/rapidwright/design/SiteConfig.java

```diff
@@ -42,11 +42,11 @@ public class AttrSite {
     private Map<BEL, Map<String, BELAttr>> belAttrs;
 
 
-    public static AttrSite getAttrSite(Site site, SiteTypeEnum type) {
-        return new AttrSite(site, type);
+    public static SiteConfig getAttrSite(Site site, SiteTypeEnum type) {
```

Comment:
```suggestion
    public static SiteConfig createSiteConfig(Site site, SiteTypeEnum type) {
```

`getSiteConfig` implies get-if-exists behaviour, whereas it looks like this is all about creating new instances?

Reply:
Thanks for catching this, yes create is more appropriate.